src/WebJobs.Script/DependencyInjection/ScriptStartupTypeLocator.cs

```diff
@@ -101,9 +102,10 @@ public async Task<IEnumerable<Type>> GetExtensionsStartupTypesAsync()
                     }
                     isPrecompiledFunctionApp = isPrecompiledFunctionApp || functionMetadata.Language == DotNetScriptTypes.DotNetAssembly;
                 }
+
+                isDotnetIsolatedApp = IsDotnetIsolatedApp(functionMetadataCollection, SystemEnvironment.Instance);
```

Comment:
Can we use the `IEnvironment` abstraction here? Rather than hardcoding `SystemEnvironment`?

Reply:
Also, why did this check move inside the `bundleConfigured` check?

Reply:
I needed the `functionMetadataCollection` which is only available inside the if block . `isDotnetIsolatedApp` is only used if bundle is configured same as `isPrecompiledFunctionApp` so moving it inside would not break the logic.

Reply:
I don't think IEnvironment abstraction is available to `ScriptStartupTypeLocator`

Reply:
This reads a little confusingly b/c if `!bundlesConfigured`... then `isDotnetIsolatedApp` will always be `false` since we never evaluate it, which isn't really the case.

I think it's the variable naming that's throwing me off. 

Can you change that name to something like `workerSupportsBundles`? That may make things read better... and it'd basically evaluate with this...

` workerSupportsBundles = !IsDotnetIsolatedApp(functionMetadataCollection, SystemEnvironment.Instance);`

Reply:
sure

Reply:
Discussed this with @brettsam offline. Going to merge this PR and follow with the above suggested changes in another PR.